gammapy/analysis/config.py

```diff
@@ -0,0 +1,189 @@
+from astropy.coordinates import Angle
+from astropy.time import Time
+from astropy.units import Quantity
+from gammapy.utils.scripts import make_path, read_yaml
+from pydantic import BaseModel, FilePath, validator
+from pydantic.utils import deep_update
+from pathlib import Path
+from typing import List
+from enum import Enum
+import yaml
+
+__all__ = ["Config", "General"]
+
+
+class AngleType(Angle):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return AngleType(v)
+
+
+class EnergyType(Quantity):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        assert isinstance(Quantity(v).to("erg"), Quantity)
+        return EnergyType(v)
+
+
+class TimeType(Time):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return TimeType(v)
+
+
+class FrameEnum(str, Enum):
+    icrs = 'icrs'
+    galactic = 'galactic'
+
+
+class BackgroundMethodEnum(str, Enum):
+    reflected = 'reflected'
+
+
+class GammapyBaseModel(BaseModel):
+    class Config:
+        validate_assignment = True
+        extra = 'forbid'
+        json_encoders = {
+            AngleType: lambda v: f"{v.value} {v.unit}",
+            EnergyType: lambda v: f"{v.value} {v.unit}",
+            TimeType: lambda v: f"{v.value}",
+        }
+
+    @classmethod
+    def from_yaml(cls, filename):
+        config = read_yaml(filename)
+        return Config(**config)
+
+    def to_yaml(self):
+        return yaml.dump(yaml.safe_load(self.json()))
+
+    def update_from_dict(self, other):
+        data = deep_update(self.dict(exclude_defaults=True), other.dict(exclude_defaults=True))
+        return Config(**data)
+
+
+class Skydir(GammapyBaseModel):
+    frame: FrameEnum = FrameEnum.icrs
+    lon: AngleType = AngleType("83.633 deg")
```

Comment:
Please rename to `SkyCoordType`, for naming consistency with the other types above.

The Crab as default position is no good. Last week I made the mistake myself to use a config template and get a source position which of course wasn't my source position. Can you just remove the default here for now?
If it is needed, then the default should be put where `SkyCoord` is defined, with a `#TODO: try to remove the default`.

Reply:
Ok.
This means that ckecking that `skydir` has been defined should be done elsewhere, no?